src/host/ledger.h

```diff
@@ -1107,28 +1114,37 @@ namespace asynchost
         });
 
       DISPATCHER_SET_MESSAGE_HANDLER(
-        disp, consensus::ledger_get, [&](const uint8_t* data, size_t size) {
-          auto [idx, purpose] =
-            ringbuffer::read_message<consensus::ledger_get>(data, size);
+        disp,
+        consensus::ledger_get_range,
+        [&](const uint8_t* data, size_t size) {
+          auto [from_idx, to_idx, purpose] =
+            ringbuffer::read_message<consensus::ledger_get_range>(data, size);
 
-          if (is_in_committed_file(idx))
+          if (is_in_committed_file(to_idx))
           {
             // Start an asynchronous job to do this, since it is committed and
             // can be accessed independently (and in parallel)
             uv_work_t* work_handle = new uv_work_t;
 
             {
-              auto data = new AsyncLedgerGet;
-              data->ledger = this;
-              data->idx = idx;
-              data->result_cb =
-                [this, idx = idx, purpose = purpose](auto&& entry, int status) {
-                  // NB: Even if status is cancelled (and entry is empty), we
-                  // want to write this result back to the enclave
-                  write_ledger_get_response(idx, std::move(entry), purpose);
-                };
-
-              work_handle->data = data;
+              auto job = new AsyncLedgerGet;
+              job->ledger = this;
+              job->from_idx = from_idx;
+              job->to_idx = to_idx;
+              job->result_cb = [this,
+                                from_idx = from_idx,
+                                to_idx = to_idx,
+                                purpose = purpose](auto&& entry, int status) {
+                // NB: Even if status is cancelled (and entry is empty), we
+                // want to write this result back to the enclave
+                write_ledger_get_range_response(
+                  from_idx,
+                  to_idx,
+                  read_framed_entries(from_idx, to_idx),
+                  purpose);
+              };
+
+              work_handle->data = job;
```

Comment:
work_handle presumably deletes the job at some point?

Reply:
Yes, this is deleted in `on_ledger_get_async_complete`. This is just renaming `data` to `job`, to avoid shadowing the existing `data` variable.